tensordict/nn/distributions/composite.py

```diff
@@ -253,47 +269,105 @@ def log_prob(
 
     def log_prob_composite(
-        self, sample: TensorDictBase, include_sum=True
+        self,
+        sample: TensorDictBase,
+        *,
+        include_sum: bool | None = None,
+        inplace: bool | None = None,
     ) -> TensorDictBase:
-        """Writes a ``<sample>_log_prob`` entry for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob.
+        """Computes the log-probability of each component in the input sample and return a TensorDict with individual log-probabilities.
+
+        Args:
+            sample (TensorDictBase): The input sample to compute the log probabilities for.
 
-        This method is called by the :meth:`~.log_prob` method when ``self.aggregate_probabilities`` is ``False``.
+        Keyword Args:
+            include_sum (bool, optional): Whether to include the summed log-probability in the output TensorDict.
+                Defaults to ``self.inplace`` which is set through the class constructor (``True`` by default).
+
+                .. warning:: The default value of ``include_sum`` will switch to ``False`` in v0.9 in the constructor.
+
+            inplace (bool, optional): Whether to update the input sample in-place or return a new TensorDict.
+                Defaults to ``self.inplace`` which is set through the class constructor (``True`` by default).
+
+                .. warning:: The default value of ``inplace`` will switch to ``False`` in v0.9 in the constructor.
+
+        Returns:
+            TensorDictBase: A TensorDict containing the individual log-probabilities for each component in the input sample,
+                along with a "sample_log_prob" entry containing the summed log-probability if `include_sum` is True.
         """
-        slp = 0.0
+        if include_sum is None:
+            include_sum = self.include_sum
+
+        if include_sum is None:
+            include_sum = True
+            warnings.warn(
+                "`include_sum` wasn't set when building the `CompositeDistribution` or when calling log_prob_composite. "
+                "The current default is ``True`` but from v0.9 it will be changed to ``False``. Please adapt your call to `log_prob_composite` accordingly.",
+                category=DeprecationWarning,
+            )
+        if inplace is None:
+            inplace = self.inplace
+        if inplace is None:
+            inplace = True
+            warnings.warn(
+                "`inplace` wasn't set when building the `CompositeDistribution` or when calling log_prob_composite. "
+                "The current default is ``True`` but from v0.9 it will be changed to ``False``. Please adapt your call to `log_prob_composite` accordingly.",
+                category=DeprecationWarning,
+            )
+        if include_sum:
+            slp = 0.0
         d = {}
         for name, dist in self.dists.items():
             d[_add_suffix(name, "_log_prob")] = lp = dist.log_prob(sample.get(name))
```

Comment:
Question: if I am to use `include_sum=False` to go through a custom aggregation downstream, should I also be in charge of setting the key for the log-probs -- instead of the default `_add_suffix(name, "_log_prob")` ?

Reply:
if you want different names and not the sum, I guess you could do `inplace=False`, `include_sum=False` and then rename?
Or we could allow you to pass a function that maps the name to a new log-prob name?
Open to suggestions